atlasdb-jdbc/src/main/java/com/palantir/atlasdb/keyvalue/jdbc/JdbcKeyValueService.java

```diff
@@ -164,7 +168,8 @@ public static JdbcKeyValueService create(JdbcKeyValueConfiguration config) {
         DataSource dataSource = dataSourceConfig.createDataSource();
         Settings settings = new Settings();
         settings.setRenderNameStyle(RenderNameStyle.AS_IS);
-        final JdbcKeyValueService kvs = new JdbcKeyValueService(config.getTablePrefix(), sqlDialect, dataSource, settings);
+        final JdbcKeyValueService kvs = new JdbcKeyValueService(settings, config.getTablePrefix(), sqlDialect,
```

Comment:
I would bias to resolving in the create method and passing params to the constructor. We will be saving just one parameter here but the constructor loses information that batchSize and table prefix are fields of the KVS.